gitlab/resource_gitlab_project_test.go

```diff
@@ -197,7 +197,75 @@ func TestAccGitlabProject_initializeWithReadme(t *testing.T) {
 				Config: testAccGitlabProjectConfigInitializeWithReadme(rInt),
 				Check: resource.ComposeTestCheckFunc(
 					testAccCheckGitlabProjectExists("gitlab_project.foo", &project),
-					testAccCheckGitlabProjectInitializeWithReadme(&project, &testAccGitlabProjectExpectedAttributes{
+					testAccCheckGitlabProjectDefaultBranch(&project, &testAccGitlabProjectExpectedAttributes{
+						DefaultBranch: "master",
+					}),
+				),
+			},
+		},
+	})
+}
+
+func TestAccGitlabProject_templateName(t *testing.T) {
+	var project gitlab.Project
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabProjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGitlabProjectConfigTemplateName(rInt),
+				Check: resource.ComposeTestCheckFunc(
```

Comment:
This `Check` step is a bit confusing, since it only checks that the project exists with name `"template-name"` and the default branch is master. Could you add an additional check that verifies that the project was indeed created with attribute `template_name = "rails"`?

Reply:
Like `initialize with readme`, init with a template is 'stateless'. So don't had the information that the project was created with a template.

Reply:
Hm okay. In that case would you mind sanity checking this then? By TF creating a project based on a template name, custom template, and project template ID, then manually checking that the template was applied?

Reply:
To resolve this, I test that README.md is present, should be enough to ensure that our repo contains any template type